homeassistant/components/vevor_heater/sensor.py

```diff
@@ -0,0 +1,143 @@
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import VevorHeaterUpdateCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS_MAPPINGS: dict[str, SensorEntityDescription] = {
+    "operational_mode": SensorEntityDescription(
+        key="operational_mode",
+        name="Operational Mode",
+        device_class=SensorDeviceClass.ENUM,
+    ),
+    "operational_status": SensorEntityDescription(
+        key="operational_status",
+        name="Operational Status",
+        device_class=SensorDeviceClass.ENUM,
+    ),
+    "input_voltage": SensorEntityDescription(
+        key="input_voltage",
+        name="Source Voltage",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+    ),
+    "elevation": SensorEntityDescription(
+        key="elevation",
+        name="Elevation",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DISTANCE,
+        native_unit_of_measurement=UnitOfLength.METERS,
+    ),
+    "target_temperature": SensorEntityDescription(
+        key="target_temperature",
+        name="Target temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+    "target_power_level": SensorEntityDescription(
+        key="target_power_level",
+        name="Target Power Level",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    "current_power_level": SensorEntityDescription(
+        key="current_power_level",
+        name="Current Power Level",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    "combustion_temperature": SensorEntityDescription(
+        key="combustion_temperature",
+        name="Combustion Chamber Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+    "room_temperature": SensorEntityDescription(
+        key="room_temperature",
+        name="Room Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+    "error:": SensorEntityDescription(
+        key="error",
+        name="Error",
+        device_class=SensorDeviceClass.ENUM,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Vevor BLE heater sensors."""
+
+    coordinator: VevorHeaterUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities = []
+    _LOGGER.debug("got sensors: %s", coordinator.data)
+    entities.extend(
```

Comment:
Looks like there is no need to make a list and a generator expression can be passed to async_add_entities

Reply:
I *definitely* didn't steal this from another integration without thinking twice about it :laughing: 

Done.

Reply:
Doesn't seem updated, did you forget to push?

Reply:
Discussed elsewhere, resolving.